types/declarative-net-request/index.d.ts

```diff
@@ -0,0 +1,20 @@
+type DNRRule = import('../../packages/ddg2dnr/lib/utils.js').DNRRule;
+type DNRRuleWithID = import('../../packages/ddg2dnr/lib/utils.js').DNRRuleWithID;
```

Comment:
I'd rather we didn't define our own type globals within .d.ts files if we can help it. Especially as it means we can't easily move to a class to represent this data which IMO would be much cleaner/simpler.

Is `DNRRule` still used on it's own (in this file and elsewhere)?